include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4413,6 +4418,24 @@ struct OrtApi {
    * \since Version 1.16.
    */
   ORT_API2_STATUS(KernelContext_GetResource, _In_ const OrtKernelContext* context, _In_ int resouce_version, _In_ int resource_id, _Outptr_ void** resource);
+
+  /** \brief Set user logging function
+   *
+   * \param[in] options
+   * \param[in] user_logging_function A pointer to a logging function.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   */
+  ORT_API2_STATUS(SetUserLoggingFunction, _Inout_ OrtSessionOptions* options, OrtLoggingFunction user_logging_function);
+
+  /** \brief Set user logging parameter for user logging function
+   *
+   * \param[in] options
+   * \param[in] user_logging_param User logging parameter for user logging function.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   */
+  ORT_API2_STATUS(SetUserLoggingParam, _Inout_ OrtSessionOptions* options, void* user_logging_param);
```

Comment:
Can we have an optional user_logging_param in SetUserLoggingFunction instead of 2 separate functions?